boot/LoadReactOS.c

```diff
@@ -0,0 +1,477 @@
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   Copyright (C) 2007 Ge van Geldorp                                     *
+ *   Copyright (C) 2019 Stanislav Motylkov                                 *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "boot.h"
+#include "BootFATX.h"
+#include "memory_layout.h"
+#include <shared.h>
+
+extern grub_error_t errnum;
+
+/* Length of area to be searched for multiboot header. */
+#define MULTIBOOT_SEARCHAREA_LEN        8192
```

Comment:
Ideally I'd like to see lots of this multiboot stuff refactored out, as it could be re-used for other multiboot-compliant binaries as well, but we can save that for a follow-up PR.

Reply:
Also see this resolved discussion: https://github.com/XboxDev/cromwell/pull/27#discussion_r359862815.
I resolved that earlier multiboot discussion because it's out of scope for this PR and probably not necessary either. there's no immediate gain from generalizing multiboot due to lack of OS.

*(Meanwhile factoring out Linux / reactOS common code removes maintenance overhead; see https://github.com/XboxDev/cromwell/pull/27#discussion_r360167001 for that)*